use standard makefile

staebler opened the pull request:
Use the makefile from https://github.com/openshift/build-machinery-go

staebler commented:
This PR builds on #935.

/hold

joelddiaz commented on Dockerfile:
```diff
@@ -17,8 +15,8 @@ RUN if ! rpm -q libvirt-devel; then yum install -y libvirt-devel && yum clean al
-COPY --from=builder /go/src/github.com/openshift/hive/bin/hive-operator /opt/services
-COPY --from=builder /go/src/github.com/openshift/hive/bin/hive-apiserver /opt/services/
+COPY --from=builder /go/src/github.com/openshift/hive/bin/operator /opt/services/hive-operator
+#COPY --from=builder /go/src/github.com/openshift/hive/bin/hive-apiserver /opt/services/
```
no need to keep the old copy command around

staebler replied:
This is an oversight where I failed to put back the code for making the hiveapiserver. This line does need to be uncommented and kept--along with more work in the makefile.

staebler replied:
This has been resolved now. I created a Makefile in v1alpha1apiserver to build the apiserver. The main Makefile will call into the apiserver Makefile for build.

I did not add generate and test code to the apiserver Makefile. I did not think it was worth the effort since (1) the apiserver code should not be changing and (2) the apiserver code will be removed soon.

joelddiaz commented on Dockerfile.dev:
```diff
@@ -9,10 +9,10 @@ RUN yum install -y libvirt-devel && yum clean all && rm -rf /var/cache/yum/*
 ADD bin/manager /opt/services/
 ADD bin/hiveutil /usr/bin/
-ADD bin/hive-apiserver /opt/services
+#ADD bin/hive-apiserver /opt/services
```
stray commented line?

staebler replied:
Also something that needs to be restored.

joelddiaz commented on docs/developing.md:
```diff
@@ -159,10 +161,11 @@ make run-operator
- 2. Deploy with your custom image: `$ DEPLOY_IMAGE=quay.io/{username}/hive:latest make deploy`
- 3. After code changes you need to rebuild the Hive images as mentioned in step 1.
- 4. Delete the running Hive pods using following command, so that the new pods will be running using the latest images built in the previous step.
+ 1. Build a custom Hive image from your current working dir: `$ IMG=quay.io/{username}/hive:latest make image-hive`
```
any particular reason to have every step listed as step number `1.`. It works AFAICT, but the rest of the docs don't use this numbering system.

staebler replied:
If everything is number `1.`, then other lines do not need to be updated when lines are added or removed.

staebler commented:
/hold cancel

#935 has merged.

staebler commented:
/uncc @LalatenduMohanty 

/cc @dgoodwin

dgoodwin commented on docs/developing.md:
````diff
@@ -108,7 +108,9 @@ go install github.com/cloudflare/cfssl/cmd/cfssl
 
 ```bash
-IMG=172.17.0.1:5000/hive:latest make docker-dev-push
+make build
````
This was a dependency of docker-dev-push, is it now not really feasible to continue that with image-hive-dev? I can live with it if not, and from the looks of how this works now it feels non-trivial to add.

staebler replied:
I can put the `docker-dev-push` target back.

staebler commented on docs/developing.md:
```diff
@@ -108,7 +108,9 @@ go install github.com/cloudflare/cfssl/cmd/cfssl
-IMG=172.17.0.1:5000/hive:latest make docker-dev-push
+make build
+IMG=172.17.0.1:5000/hive:latest make build-hive-dev 
+IMG=172.17.0.1:5000/hive:latest make docker-push
```
That is what it does in master, too.

```
[staebler@localhost hive (master)]$ make docker-push
docker push hive-controller:latest
The push refers to repository [docker.io/library/hive-controller]
79d5f86dd345: Preparing 
dfd7bfa09eb9: Preparing 
77b174a6a187: Preparing 
denied: requested access to the resource is denied
make: *** [Makefile:240: docker-push] Error 1
```

dgoodwin commented:
Needs a merge conflict resolved, otherwise lgtm.

staebler commented:
> Needs a merge conflict resolved, otherwise lgtm.

Rebased. I also added one more commit to handle the different ldflags used by the standard makefile for handling version info.

dgoodwin commented:
/lgtm
